src/allocation_engine/README.md

```diff
@@ -0,0 +1,170 @@
+# Design
+
+## Allocation engine
+
+This implementation uses an interval tree that is specialised for allocation of
+memory-mapped I/O and port I/O address space. The fields of the strctures
+defined will have semantic meaning for this context (e.g node state to indicate
+if a node in the tree is assigned or not to a device).
+
+We offer three options for placing a memory slot in the managed address space:
```

Comment:
FirstMatch and LastMatch don't need a position argument. Would it make sense then to have `ExactMatch` define the range that needs to be allocated? This will enable us to have the same signature for allocation functions because the difference in parameter is captured in the allocationa_policy field.

Reply:
I really like this suggestion.